src/include/hardware.h

```diff
@@ -73,6 +73,47 @@
 #define WT_WRITE_ONCE(v, val) WT_RELEASE_WRITE_WITH_BARRIER(v, val)
 #endif
 
+/*
+ * The below assembly implements the read-acquire semantic. The if branches get removed at compile
```

Comment:
Most of this comment only applies to the LDAPR implementation. I'd prefer if this top level comment explains the behaviour of `ACQUIRE_READ` and how it differs from `WITH_BARRIER`, and then this current comment goes inside the `#HAVE_RCPC` macro

Reply:
I think we should explain that in the gcc.h header comment which needs updating in: WT-10967

Reply:
Do you want a fix me?

Reply:
20359cc56